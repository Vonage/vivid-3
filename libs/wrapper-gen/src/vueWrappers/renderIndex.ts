import { ComponentDef } from '../common/ComponentDef';
import { wrappedComponentName } from './name';

export default function renderIndex(components: ComponentDef[]) {
	return components
		.map((component) => wrappedComponentName(component))
		.map(
			(component) => `export { default as ${component} } from './${component}';`
		)
		.join('\n');
}
