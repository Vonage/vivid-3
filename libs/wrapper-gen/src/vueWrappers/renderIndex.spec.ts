import renderIndex from './renderIndex';
import {
	exampleComponent,
	minimalComponent,
} from '../__fixtures__/componentDefs';

describe('renderIndex', () => {
	it('should render a barrel file that exports all components', () => {
		expect(renderIndex([minimalComponent, exampleComponent])).toMatchSnapshot();
	});
});
