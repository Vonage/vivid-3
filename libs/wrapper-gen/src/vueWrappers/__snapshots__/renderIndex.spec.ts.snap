// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renderIndex > should render a barrel file that exports all components 1`] = `
"export { default as VExampleComponent } from './VExampleComponent';
export { default as VExampleComponent } from './VExampleComponent';"
`;
